app/actions/websocket/channel.test.ts

```diff
@@ -0,0 +1,577 @@
+            },
+            broadcast: {
+                channel_id: channelId,
+                userId,
+                user_id: userId,
+            },
+        };
+        DatabaseManager.getActiveServerUrl = jest.fn().mockResolvedValue(serverUrl);
+    });
+
+    afterEach(async () => {
+        await DatabaseManager.destroyServerDatabase(serverUrl);
+        jest.restoreAllMocks();
+    });
+
+    describe('handleChannelCreatedEvent', () => {
+        it('should handle channel created event', async () => {
+            (EphemeralStore.creatingChannel as boolean) = false;
+            (getChannelById as jest.Mock).mockResolvedValue(null);
+            (fetchMyChannel as jest.Mock).mockResolvedValue({channels: [{}], memberships: [{}]});
+            (prepareMyChannelsForTeam as jest.Mock).mockResolvedValue([[]]);
+            (addChannelToDefaultCategory as jest.Mock).mockResolvedValue({models: []});
+
+            await handleChannelCreatedEvent(serverUrl, msg);
+
+            expect(getChannelById).toHaveBeenCalled();
+            expect(fetchMyChannel).toHaveBeenCalledWith(serverUrl, teamId, channelId, true);
+            expect(prepareMyChannelsForTeam).toHaveBeenCalled();
+            expect(addChannelToDefaultCategory).toHaveBeenCalled();
+        });
+
+        it('should handle channel created event - no channels', async () => {
+            (EphemeralStore.creatingChannel as boolean) = false;
+            (getChannelById as jest.Mock).mockResolvedValue(null);
+            (fetchMyChannel as jest.Mock).mockResolvedValue({channels: null, memberships: [{}]});
+
+            await handleChannelCreatedEvent(serverUrl, msg);
+
+            expect(getChannelById).toHaveBeenCalled();
+            expect(fetchMyChannel).toHaveBeenCalledWith(serverUrl, teamId, channelId, true);
+            expect(prepareMyChannelsForTeam).not.toHaveBeenCalled();
+        });
+
+        it('should handle channel created event - nothing to prepare', async () => {
+            (EphemeralStore.creatingChannel as boolean) = false;
+            (getChannelById as jest.Mock).mockResolvedValue(null);
+            (fetchMyChannel as jest.Mock).mockResolvedValue({channels: [{}], memberships: [{}]});
+            (prepareMyChannelsForTeam as jest.Mock).mockResolvedValue([]);
+
+            await handleChannelCreatedEvent(serverUrl, msg);
+
+            expect(getChannelById).toHaveBeenCalled();
+            expect(fetchMyChannel).toHaveBeenCalledWith(serverUrl, teamId, channelId, true);
+            expect(prepareMyChannelsForTeam).toHaveBeenCalled();
+            expect(addChannelToDefaultCategory).not.toHaveBeenCalled();
+        });
+
+        it('should return if creatingChannel is true', async () => {
+            (EphemeralStore.creatingChannel as boolean) = true;
+
+            await handleChannelCreatedEvent(serverUrl, msg);
+
+            expect(getChannelById).not.toHaveBeenCalled();
```

Comment:
We are checking here for a implementation detail. We should be checking the effect of the function instead (whether the channel gets added to the database, for example).